[expo-blob] Added ExpoBlob package (#38366)
expo/expo: docs/pages/versions/unversioned/sdk/blob.mdx

---
title: Blob
description: A web standards-compliant Blob implementation for React Native.
sourceCodeUrl: 'https://github.com/expo/expo/tree/main/packages/expo-blob'
packageName: 'expo-blob'
platforms: ['android', 'ios', 'web']
---

import APISection from '~/components/plugins/APISection';
import { APIInstallSection } from '~/components/plugins/InstallSection';

> import { SnackInline } from '~/ui/components/Snippet';

Reviewer: ```suggestion
```

---
title: Blob
description: A web standards-compliant Blob implementation for React Native.
sourceCodeUrl: 'https://github.com/expo/expo/tree/main/packages/expo-blob'
packageName: 'expo-blob'

> iconUrl: '/static/images/packages/expo-file-system.png'

Reviewer 2: ```suggestion
```

There's no icon URL as of yet for this package/page.

platforms: ['android', 'ios', 'web']
---

import APISection from '~/components/plugins/APISection';
import { APIInstallSection } from '~/components/plugins/InstallSection';
import { SnackInline } from '~/ui/components/Snippet';

> `expo-blob` provides a web standards-compliant Blob implementation for React Native that offers superior performance and reliability compared to the implementation exported from `react-native`. Unlike Blob from `react-native`, which has limitations with the `slice()` method and other Web API features, `expo-blob` provides a complete, standards-compliant implementation that works consistently across all platforms.

Reviewer 2: ```suggestion
`expo-blob` provides a web standards-compliant Blob implementation for React Native that offers superior performance and works consistently across all platforms. It is more reliable compared to the implementation exported from `react-native`, unlike Blob from `react-native`, which has limitations with the `slice()` method and other Web API features.
```

Last line was repetitive since the same context is shared in the first line. I'd suggest rewriting this introduction paragraph.

## Installation

<APIInstallSection />

## Usage

> ### Basic Blob Creation

Reviewer 2: ```suggestion
### Basic Blob creation
```

As per our writing style guide, for section headings we prefer sentence case: https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#headings.

```typescript
import { Blob } from 'expo-blob';

// Create an empty blob
const emptyBlob = new Blob();

// Create a blob from text
const textBlob = new Blob(['Hello, World!'], { type: 'text/plain' });

// Create a blob from binary data
const binaryBlob = new Blob([new Uint8Array([1, 2, 3, 4])], {
  type: 'application/octet-stream',
});

// Create a blob from mixed content
const mixedBlob = new Blob(
  [
    'Text content',
    new Uint8Array([65, 66, 67]), // ABC in ASCII
    'More text',
  ],
  { type: 'text/plain' }
);
```

> ### Blob Properties

Reviewer 2: ```suggestion
### Blob properties
```

```typescript
const blob = new Blob(['Hello, World!'], { type: 'text/plain' });

console.log(blob.size); // 13 (bytes)
console.log(blob.type); // "text/plain"
```

> ### Reading Blob Content

Reviewer 2: ```suggestion
### Reading Blob content
```
